src/clj_kondo/impl/analyzer.clj

```diff
@@ -941,6 +941,35 @@
        ctx
        (node->line (:filename ctx) expr :syntax (format "%s binding vector requires exactly 2 forms" form-name))))))
 
+(defn condition-always-true-linter
+  [ctx expr]
+  (findings/reg-finding! ctx (assoc (meta expr)
+                                    :filename (:filename ctx)
+                                    :message "Condition always true"
+                                    :type :condition-always-true)))
+
+(defn analyze-condition
+  [ctx condition]
+  (let [ctx (update ctx :arg-types #(or % (atom [])))
+        condition (assoc condition :condition true)
+        analyzed (doall (analyze-expression** ctx condition))
+        t (tag condition)
+        loc (select-keys (if (identical? :quote t)
+                           (meta (first (:children condition)))
+                           (meta condition))
+                         [:row :col :end-row :end-col])]
```

Comment:
This is a hack but idk how to get around it. We need some way of finding out which `:arg-types` entry is for our condition, and the the quoted list's metadata is attached to the child node.

Reply:
`:arg-types` is a vector that tracks the location and type of analyzed forms. Because it's a vector, we have to go from our condition form to the arg-type form in some way. We can't just go by the shape cuz there could be identical forms, so we gotta go by position. However, the location metadata on a quoted form (pre-analyzed) will be different than the location metadata on that form's arg-type because it uses the actual form's location:

```
'foo
^-- (meta condition)
 ^-- :arg-type
```

To solve this, I "unwrap" the quote so that the `loc` I'm looking for is the `foo` object itself.